codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -178,15 +204,22 @@ public float windowUpdateRatio(Http2Stream stream) throws Http2Exception {
     @Override
     public void receiveFlowControlledFrame(ChannelHandlerContext ctx, Http2Stream stream, ByteBuf data,
             int padding, boolean endOfStream) throws Http2Exception {
+        this.ctx = checkNotNull(ctx, "ctx");
```

Comment:
Hum I'm not sure if one way is better/different from another but should we save this reference in the `consumeBytes` method, because that is what it is used for?